File: tools/is_tty_test.ts
// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
import { args, exit, isTTY } from "deno";

const name = args[1];
const test = {
  stdin: () => {
    console.log(isTTY().stdin);
  },
  stdout: () => {
    console.log(isTTY().stdout);
  },
  stderr: () => {
    console.log(isTTY().stderr);
  }
}[name];

if (!test) {
  console.log("Unknown test:", name);
  exit(1);
}

test();

File: tools/is_tty_test.py
#!/usr/bin/env python
# Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
import os
import unittest
from sys import stdin

from test_util import DenoTestCase, run_tests
from util import tty_capture

IS_TTY_TEST_TS = "tests/is_tty.ts"


@unittest.skipIf(os.name == 'nt', "Unable to test tty on Windows")
class TestIsTty(DenoTestCase):
    def test_is_tty(self):
        cmd = [self.deno_exe, "run", IS_TTY_TEST_TS]
        code, stdout, _ = tty_capture(cmd, b'')
        assert code == 0
        assert str(stdin.isatty()).lower() in stdout


if __name__ == "__main__":
    run_tests()
